src/DonorDashboards/resources/js/app/components/subscription-manager/index.js

```diff
@@ -17,19 +17,53 @@ const SubscriptionManager = ( { id, subscription } ) => {
 	const [ paymentMethod, setPaymentMethod ] = useState( null );
 	const [ isUpdating, setIsUpdating ] = useState( false );
 	const [ updated, setUpdated ] = useState( true );
+	const [ cardControl, setCardControl ] = useState( null );
 
 	useEffect( () => {
 		setUpdated( false );
 	}, [ amount, paymentMethod ] );
 
 	const handleUpdate = async() => {
-		// Save with REST API
+		if ( isUpdating ) {
+			return;
+		}
+
 		setIsUpdating( true );
+
+		let paymentMethodStripe;
+
+		if ( cardControl ) {
```

Comment:
@alaca  `handleUpdate` becomes fat after adding Stripe payment method creation logic and in the future chances are higher that this function will become bigger when we will add more features to subscription edit.

Can we develop something which keeps component-related logic in component but form submission still have control of payment method?

Reply:
It's not that easy. `handleUpdate` handles updates of all subscriptions regardless of the gateway. In order to handle  Stripe subscriptions, `StripeCardControl` component must be wrapped with the `Elements` component from `@stripe/react-stripe-js`. Then you have access to the `useStripe` function which is needed to handle Stripe subscriptions. In other words, this logic must be contained in the `handleUpdate` function because this function initializes the update and it depends on additional request to get the payment id which is used to update the Stripe subscription.

Reply:
I see what you both mean. Ideally, we'd have something like:
```jsx
const gateway =  getGatewayController(subscription.gateway.id);

gateway.handleSubscriptionUpdate(amount, cardDetails);
```

In which case each gateway would be responsible for handling the submission and update. But I get that this is a considerable refactor. As a side note, we need to start building out more first-class systems for gateways to handle themselves.

Anyway, the purpose of this PR is to get things working. If there isn't a somewhat timely way of pulling this off — at your discretion, @alaca  — then proceed.

Reply:
The only thing that differs now between the gateways is how to get the payment method ID. @ravinderk is right, that logic should be per component. If we add 5 more gateways that are working similar to Stripe, we will end up with a mess that will be hard to maintain. So yeah, we should refactor this. 

Using a controller might not be a good approach as handling the gateway like Stripe heavily depends on the card input component. To be precise, you can't do anything with Stripe outside `StripeCardControl` or without exposing the `stripe` object. But in order to get the `stripe` object in the first place, we have to use `useStripe` function which has to be wrapped with `Elements` component from `@stripe/react-stripe-js`.

That's why I think using controllers might not be a good solution here, because they will have to encapsulate all the mess we have now, and even more.

I'll try to solve this using `refs`.

Reply:
@JasonTheAdams I think I'm heading in the "right" direction now. At least it is better than what we had before. 

Now all the side effects are handled by the child component. By using `refs` and `useImperativeHandle`, the `getPaymentMethod` is now accessible from parent component. All the logic to handle specific gateway is now contained within the child component. The only thing I don't like is that we are using a lot of prop drilling to pass the ref, and that we are not using TypeScript. If we used TypeScript here, we could use an interface for these kinds of stuff.

Reply:
Great work, @alaca! Yeah, prop drilling isn't fun, but if it's only 1-2 layers, I think it's fine. You're right that TS would help here, and perhaps at some point we'll begin the switch. I'd want there to be some collaboration from the dev team to determine when and how the change would be made.

This is good enough for now. Let's test this with the Recurring PR, get these merged, and out the door.

Reply:
@alaca All recurring pull requests has been merged. You can test this against recurring `develop` branch.

Reply:
@alaca Alright, i will take care of Paypal Pro and Authorize.Net

Reply:
@ravinderk updating stripe subscriptions work! If you can confirm the same for the paypal and authorize, then this is good to go.